### PR TITLE
update: updated plugin list command

### PR DESCRIPTION
**Pull request by patrickzheng200**
This PR mainly fixed the plugin list command.
Previously, if there's an error on getting a plugin, only the error will be printed out. 
In this PR, on error, the pluginName will also be printed out along with the error.
(This PR also includes some of the minor cleanups of notation CLI.)

Signed-off-by: Patrick Zheng <patrickzheng@microsoft.com>

### CODE REVIEW

File: cmd/notation/plugin.go [hunk 1]
```diff
@@ -42,5 +42,5 @@ func listPlugins(command *cobra.Command) error {
 		errors = append(errors, err)
 		plugins = append(plugins, pl)
 	}
-	return ioutil.PrintPlugins(command.Context(), os.Stdout, plugins, errors)
+	return ioutil.PrintPlugins(command.Context(), os.Stdout, pluginNames, plugins, errors)
```
[the remark this answers is not in the document]
Comment (patrickzheng200) [reply cmd/notation/plugin.go]: Code updated.

File: cmd/notation/plugin.go [hunk 2]
```diff
@@ -30,17 +32,30 @@ func pluginListCommand() *cobra.Command {
...
+
+	tw := tabwriter.NewWriter(os.Stdout, 0, 0, 3, ' ', 0)
+	fmt.Fprintln(tw, "NAME\tDESCRIPTION\tVERSION\tCAPABILITIES\tERROR\t")
 	for _, n := range pluginNames {
-		pl, err := mgr.Get(command.Context(), n)
-		errors = append(errors, err)
-		plugins = append(plugins, pl)
+		pl, err = mgr.Get(command.Context(), n)
+		metaData := &proto.GetMetadataResponse{}
+		if pl != nil {
```
Comment (shizhMSFT) [inline cmd/notation/plugin.go:50]: Instead of checking `pl`, always check `err`.

**Review by shizhMSFT — APPROVED** [review]
LGTM

File: cmd/notation/plugin.go [hunk 3]
```diff
@@ -30,17 +32,30 @@ func pluginListCommand() *cobra.Command {
...
+	fmt.Fprintln(tw, "NAME\tDESCRIPTION\tVERSION\tCAPABILITIES\tERROR\t")
 	for _, n := range pluginNames {
-		pl, err := mgr.Get(command.Context(), n)
-		errors = append(errors, err)
-		plugins = append(plugins, pl)
+		pl, err = mgr.Get(command.Context(), n)
+		metaData := &proto.GetMetadataResponse{}
+		if err == nil {
+			req := &proto.GetMetadataRequest{}
+			resp, err = pl.GetMetadata(command.Context(), req)
```
Comment (priteshbandi) [inline cmd/notation/plugin.go:52]:
```suggestion
			resp, err = pl.GetMetadata(ctx, &proto.GetMetadataRequest{})
```

File: cmd/notation/plugin.go [hunk 4]
```diff
@@ -30,17 +32,30 @@ func pluginListCommand() *cobra.Command {
 }
 
 func listPlugins(command *cobra.Command) error {
+	var err error
+	var pl plugin.Plugin
+	var resp *proto.GetMetadataResponse
+
```
Comment (priteshbandi) [inline cmd/notation/plugin.go:38]: nit: declare variable close to the place where they are actually getting used.

**Review by priteshbandi — APPROVED** [review]
LGTM with suggestions

File: cmd/notation/plugin.go [hunk 5]
```diff
@@ -30,17 +32,29 @@ func pluginListCommand() *cobra.Command {
 }
 
 func listPlugins(command *cobra.Command) error {
+	var err error
```
Comment (shizhMSFT) [inline cmd/notation/plugin.go:35]: This line is redundant.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**